### PR TITLE
Make SystemInfo a Resource

### PR DESCRIPTION
**Pull request by IceSentry**
# Objective

We already collect a lot of system information on startup when possible but we don't make this information available. With the upcoming work on a diagnostic overlay it would be useful to be able to display this information.

## Solution

Make the already existing SystemInfo a Resource

---

## Changelog

Added `SystemInfo` Resource

### CODE REVIEW

File: crates/bevy_diagnostic/src/system_information_diagnostics_plugin.rs [hunk 1]
```diff
@@ -122,6 +126,7 @@ pub mod internal {
         };
 
         info!("{:?}", info);
+        commands.insert_resource(info);
```
Comment (pablo-lua) [inline crates/bevy_diagnostic/src/system_information_diagnostics_plugin.rs:129]: Can we somehow do this at app build time? This way, we can use this struct in Startup without having to worry about system ordering at all.

**Review by pablo-lua — COMMENTED** [review]
Its good to have this available all the time, I just have a comment.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
We should be able to just initialize and log the system info using a `FromWorld` impl + `init_resource`. Like @pablo-lua says, this will allow us to dodge system ordering issues (and reduce overhead).

**Comment by IceSentry** [comment]
So, I added the FromWorld impl as suggested. I'm not sure how to handle the case where sysinfo is not supported though. right now we are kinda forced to always init the resource anyway so I created one that is just `"Unknown"` for everything. We could alternatively just not init it and hope that people querying it don't use a fallible way to access it. I think this solution reduces the chances of runtime panics at the cost of having `"Unknown"` show up in the overlay.

Another issue I'm really confused about with this is that the fps_overlay example now spawns it's UI in a different order because of my changes. It's as if it's based on timing of the startup time or something like that.

**Review by alice-i-cecile — APPROVED** [review]
Oh you're right, we really don't need the `World` there. We can actually just get away with a `Default` impl, sorry! I'm content with the other changes: "unknown" is a sensible fallback.

**Comment by IceSentry** [comment]
Here's the fix for the UI bug in the example https://github.com/bevyengine/bevy/pull/12586

I'm not entirely sure why this PR changes the spawn order but the fps_overlay ui code was wrong anyway so I don't want to look into it too much.

**Review by pablo-lua — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@IceSentry CI failures are real this time :)

**Comment by IceSentry** [comment]
Those cfg errors are really hard to detect locally 😅 

Should be good now

File: crates/bevy_diagnostic/src/system_information_diagnostics_plugin.rs [hunk 2]
```diff
@@ -24,10 +25,21 @@ impl Plugin for SystemInformationDiagnosticsPlugin {
...
 
 impl SystemInformationDiagnosticsPlugin {
+    /// Total system cpu usage in %
     pub const CPU_USAGE: DiagnosticPath = DiagnosticPath::const_new("system/cpu_usage");
+    /// Total system memory usage in %
     pub const MEM_USAGE: DiagnosticPath = DiagnosticPath::const_new("system/mem_usage");
 }
 
+#[derive(Debug, Resource)]
+pub struct SystemInfo {
```
Comment (afonsolage) [inline crates/bevy_diagnostic/src/system_information_diagnostics_plugin.rs:35]:
```suggestion
/// A resource that stores diagnostic information about the system.
/// This information can be useful for debugging and profiling purposes.
///
/// # See also
///
/// [`SystemInformationDiagnosticsPlugin`](crate::SystemInformationDiagnosticsPlugin) for more information.
pub struct SystemInfo {
```

**Review by afonsolage — COMMENTED** [review]
Since `SystemInfo` is in a public API now, it should have some level of doc comments.

**Review by james7132 — APPROVED** [review]
Please fix CI.

**Pull request merged**